lms/lmsdb/models.py

```diff
@@ -265,6 +267,27 @@ def is_user_registered(cls, user_id: int, course_id: int) -> bool:
         )
 
 
+@post_save(sender=UserCourse)
+def on_save_user_course(model_class, instance, created):
+    """Changes user's last course viewed."""
+
+    if instance.user.last_course_viewed is None:
+        instance.user.last_course_viewed = instance.course
+        instance.user.save()
+
+
+@post_delete(sender=UserCourse)
+def on_delete_user_course(model_class, instance):
+    """Changes user's last course viewed."""
+
+    if instance.user.last_course_viewed == instance.course:
+        if new_last_course_viewed := Course.fetch(instance.user).limit(1):
```

Comment:
If there are no courses -- will it return None or error?

If it's None - just do the assignment, if an error - you shouldn't use a walrus here 'cause it won't catch the exception